src/transformers/modeling_utils.py

```diff
@@ -2165,10 +2165,11 @@ def find_submodule_and_param_name(model, long_key):
                 submodule, param_name = find_submodule_and_param_name(model, k)
                 if submodule is not None:
                     param_dtype = getattr(submodule, param_name).dtype
-                    new_val = state_dict[k].to(param_dtype)
-                    if isinstance(getattr(submodule, param_name), torch.nn.Parameter):
-                        new_val = torch.nn.Parameter(new_val)
-                    setattr(submodule, param_name, new_val)
+                    if k in state_dict:
```

Comment:
This test should go above on line 2165 with a `continue` if it's not `True`, to avoid looking for the param when we don't need it.

Reply:
Updated. the only difference to your comment is `setattr(submodule, param_name, new_val)` is after the check for the key

Reply:
There is nothing on line 2165, are you sure you pushed your update? The goal is to avoid spending any time in this block (starting at `submodule, param_name = find_submodule_and_param_name(model, k)`) when there is no need to.

Reply:
Apologies, updated. No need to ugly `continue` when you can do everything with a positive conditional flow

Reply:
Prefilter?

```
keys_to_load = [k for k in loaded_state_dict_keys if k in state_dict]
```

Reply:
then why not just iterate using the `state_dict` keys as in my first proposal in the RegNet PR

Reply:
it won't be the same if `loaded_state_dict_keys` doesn't include all state_dict keys. I'm pretty sure it is right now, but it may change. Note this warning:

https://github.com/huggingface/transformers/blob/10131afc4d8ebc4e352196db8a05da6b8bef9ee7/src/transformers/modeling_utils.py#L2121

it was a quick hack to enable an urgent use so it needs to be completed to do a full support, in which case not all key from state_dict might be loaded.

Reply:
I only suggested the comprehension way as another way to avoid too much conditional nesting.

`continue` is there for this exact reason and a functional programming tool

Reply:
I suggested a simple alternative to deep conditional nesting here: https://github.com/huggingface/transformers/pull/16643#discussion_r845298572

But I'm fine with the code the way it is now as well.

Reply:
Sure, what I meant is that prefiltering is the same as just iterating the loaded state_dict keys, that is the cleanest solution